retail-managment-system/rms-ui/src/app/pages/installments/edit-installment/edit-installment.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { InstallmentRepositry } from 'src/app/domain/installment/installment.repositry';
+import { Installment } from 'src/app/domain/installment/models/installment';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { TranslateService } from '@ngx-translate/core';
+
+
+
+
+
+@Component({
+  selector: 'app-edit-installment',
+  templateUrl: './edit-installment.component.html',
+  styleUrls: ['./edit-installment.component.scss']
+})
+export class EditInstallmentComponent implements OnInit {
+  installForm!: FormGroup;
+  isButtonVisible!: boolean;
+  submit: boolean = false;
+  paymentAmount!: number;
+  paymentDate!: Date;
+ constructor(private installmentrepositry: InstallmentRepositry,
+  private formBuilder : FormBuilder,
+  private _snackBar: MatSnackBar,
+  private dialog: MatDialog,
+  private translate: TranslateService,
+  @Inject (MAT_DIALOG_DATA) public editData : Installment
+ ) { }
+
+  ngOnInit() {
+   this.installmentForm();
+   this.fetchData();
+  }
+
+  installmentForm() {
+    this.installForm = this.formBuilder.group({
+      id: [''],
+      installmentAmount:[''],
+      paymentAmount: ['' , [Validators.required]], 
+      paymentDate: [''],
+      status:['']
+    });
+  }
+
+ 
+  fetchData() {
+   if (this.editData)
+   {
+    this.installForm.controls['paymentAmount'].setValue(this.editData.paymentAmount)
+    this.installForm.controls['installmentAmount'].setValue(this.editData.installmentAmount)
+    this.paymentAmount = this.installForm.controls['paymentAmount'].value
+    this.installForm.controls['paymentDate'].setValue(this.editData.paymentDate)
+    this.paymentDate=this.installForm.controls['paymentDate'].value
+   }
+  }
+
+  updateInstallment() {
+    this.isButtonVisible = true;
+    this.submit = true;
+    this.installmentrepositry.update(this.installForm.value).subscribe(() => {
+     this.clickButton();
+      this.dialog.closeAll();
```

Comment:
use a function that you already added for close
to achieve single responsibility